packages/nextjs/pages/myNFTs.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from "react";
+import type { NextPage } from "next";
+import { useAccount } from "wagmi";
+import { MetaHeader } from "~~/components/MetaHeader";
+import { RainbowKitCustomConnectButton } from "~~/components/scaffold-eth";
+import { MyHoldings } from "~~/components/simpleNFT";
+import { useScaffoldContractWrite } from "~~/hooks/scaffold-eth";
+import { notification } from "~~/utils/scaffold-eth";
+import { ipfsClient } from "~~/utils/simpleNFT";
+import nftsMetadata from "~~/utils/simpleNFT/nftsMetadata";
+
+const MyNFTs: NextPage = () => {
+  const { address: connectedAddress, isConnected, isConnecting } = useAccount();
+  const [currentTokenMintCount, setCurrentTokenMintCount] = useState(0);
+
+  const { writeAsync: mintItem } = useScaffoldContractWrite({
+    contractName: "YourCollectible",
+    functionName: "mintItem",
+    args: [connectedAddress, ""],
+  });
+
+  const handleMintItem = async () => {
+    // circle back to the zero item if we've reached the end of the array
+    const currentTokenMetaData = nftsMetadata[currentTokenMintCount % nftsMetadata.length];
+    const notificationId = notification.loading("Uploading to IPFS");
+    try {
+      const uploadedItem = await ipfsClient.add(JSON.stringify(currentTokenMetaData));
+
+      // First remove previous loading notification and then show success notification
+      notification.remove(notificationId);
+      notification.success("Metadata uploaded to IPFS");
+
+      await mintItem({
+        args: [connectedAddress, uploadedItem.path],
+      });
+
+      setCurrentTokenMintCount(prevCount => prevCount + 1);
+    } catch (error) {
+      notification.remove(notificationId);
+      console.error(error);
+    }
+  };
+
+  return (
+    <>
+      <MetaHeader />
+      <div className="flex items-center flex-col pt-10">
+        <div className="px-5">
+          <h1 className="text-center mb-8">
+            <span className="block text-4xl font-bold">My NFTs</span>
+          </h1>
+        </div>
+      </div>
+      <div className="flex justify-center">
+        {!isConnected || isConnecting ? (
+          <RainbowKitCustomConnectButton />
+        ) : (
+          <button className="btn btn-secondary" onClick={handleMintItem}>
+            Mint Item
```

Comment:
At Readme we talk about "Mint NFT button". We should probably change text here or change Readme instead if we prefer "Mint Item"

Reply:
Yup I update it to "Mint NFT" at [a79abb4](https://github.com/scaffold-eth/se-2-challenges/pull/5/commits/a79abb413b5d90977acde4abe05a9eb7678bfcf2)